pkg/tenant/db/tenants.go

```diff
@@ -54,13 +55,40 @@ func (t *Tenant) UnmarshalValue(data []byte) error {
 	return msgpack.Unmarshal(data, t)
 }
 
+func (t *Tenant) Validate() error {
+	if ulids.IsZero(t.OrgID) {
+		return ErrMissingOrgID
+	}
+
+	if t.Name == "" {
+		return ErrMissingTenantName
+	}
+
+	if t.EnvironmentType == "" {
+		return ErrMissingEnvType
+	}
+
+	alphaNum := regexp.MustCompile(`^[A-Za-z][A-Za-z0-9]*$`)
+
+	if !alphaNum.MatchString(t.Name) {
+		return ErrValidation
+	}
+
+	return nil
```

Comment:
Thanks for adding this, I had created a story for this but I can close that now.